### PR TITLE
Fix `param_shift` for broadcasted tapes

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
The parameter-shift gradient transform did not work correctly with tapes/QNodes that already are broadcasted. An incomprehensible error was raised in this case.

**Description of the Change:**
This PR fixes this flaw, which was based on a flawed logic in determining whether an existing `batch_size` is related to the kwarg `broadcast=True`, or whether it is simply there because the original object already was broadcasted.

**Benefits:**
The bug is resolved and `param_shift` can be used on tapes and QNodes that already have broadcasted parameters. This is independent - and indeed incompatible - with the `broadcast=True` keyword argument, which determines whether the transform itself should use broadcasting.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
N/A

### CODE REVIEW

File: pennylane/gradients/parameter_shift.py [hunk 1]
```diff
@@ -690,7 +690,7 @@ def expval_param_shift(
         # If broadcast=True, g_tapes only contains one tape. If broadcast=False, all returned
         # tapes will have the same batch_size=None. Thus we only use g_tapes[0].batch_size here.
         # If no gradient tapes are returned (e.g. only unshifted term in recipe), batch_size=None
-        batch_size = g_tapes[0].batch_size if g_tapes else None
+        batch_size = g_tapes[0].batch_size if broadcast and g_tapes else None
```
Comment (rmoyard) [inline pennylane/gradients/parameter_shift.py:693]: Great catch :+1:

File: tests/returntypes/paramshift/test_parameter_shift_shot_vec_new.py [hunk 2]
```diff
@@ -615,7 +615,9 @@ class NewOp(qml.operation.Operation):
                 qml.gradients.param_shift(tape, shots=shot_vec)
 
 
-# TODO: add test class for parameter broadcasting
+# TODO: add test class for parameter broadcasting - both when using broadcasting in the transform
```
Comment (rmoyard) [inline tests/returntypes/paramshift/test_parameter_shift_shot_vec_new.py:618]: Is it planned for an upcoming PR or for this one?

**Review by rmoyard — APPROVED** [review]
Thank you @dwierichs it looks good! Do you plan to add the tests for shot vec in this PR or in an upcoming one? If not in this one, you can merge it!

**Comment by dwierichs** [comment]
I think this is out of scope for this PR, as there is a gap for shot vector + broadcasting already :)
Will merge , then!

**Pull request merged**